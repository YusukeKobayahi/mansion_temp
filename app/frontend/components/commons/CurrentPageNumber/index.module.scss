@import "~/styles/variables";
@import "~/styles/utils";

.container {
  display: flex;
}

.currentCount {
  font-size: 25px;
  color: #746744;
  font-weight: bold;
  white-space: nowrap;
}

.totalCount {
  font-size: 16px;
  color: #746744;
  font-weight: bold;
  white-space: nowrap;
  align-self: flex-end;
}
