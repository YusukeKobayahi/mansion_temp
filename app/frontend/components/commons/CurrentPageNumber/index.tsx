import styles from "~/components/commons/CurrentPageNumber/index.module.scss";

export interface CurrentPageNumberProps {
  currentPage: number;
  totalCount: number;
}

const CurrentPageNumber: React.FC<CurrentPageNumberProps> = ({
  currentPage = 1,
  totalCount,
}) => {
  return (
    <div className={styles.container}>
      <p className={styles.currentCount}>
        {currentPage * 15 < totalCount ? currentPage * 15 : totalCount} /{" "}
      </p>
      <p className={styles.totalCount}>{totalCount}ä»¶</p>
    </div>
  );
};

export default CurrentPageNumber;
