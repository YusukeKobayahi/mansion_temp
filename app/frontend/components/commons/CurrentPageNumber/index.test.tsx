import { render } from "@testing-library/react";
import CurrentPageNumber from "~/components/commons/CurrentPageNumber";

describe("components/commons/CurrentPageNumber", () => {
  it("正常に表示される", () => {
    const dummyData = {
      currentPage: 5,
      totalPages: 10,
    };
    const { asFragment } = render(<CurrentPageNumber {...dummyData} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
