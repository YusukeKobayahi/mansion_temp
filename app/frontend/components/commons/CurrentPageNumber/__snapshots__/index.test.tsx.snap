// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/commons/CurrentPageNumber 正常に表示される 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <p
      class="currentCount"
    >
       / 
    </p>
    <p
      class="totalCount"
    >
      件
    </p>
  </div>
</DocumentFragment>
`;
